datafusion/physical-plan/src/repartition/mod.rs

```diff
@@ -774,6 +775,10 @@ impl DisplayAs for RepartitionExec {
 
                 if self.preserve_order {
                     writeln!(f, "preserve_order={}", self.preserve_order)?;
+                } else if input_partition_count <= 1
```

Comment:
Since the tree format is designed to be less verbose, I personally suggest not adding the additional information to the tree format as well -- people who want more detail can use the indent format